execute/plugin.go

```diff
@@ -472,6 +482,16 @@ func (p *Plugin) Reports(
 		},
 	}
 
+	// Update cache.
+	// Note: this could wait until should transmit, but the next round may start before then.
+	// For that reason we update caches here. If there is some sort of failure to transmit,
+	// messages will be delayed until the inflight cache expires and messages are reprocessed.
+	for _, chainReport := range decodedOutcome.Report.ChainReports {
+		for _, message := range chainReport.Messages {
+			p.inflightMessageCache.MarkInflight(chainReport.SourceChainSelector, message.Header.MessageID)
+		}
+	}
```

Comment:
Reports() and Outcome() are supposed to be pure, I feel like we're straying farther from that and this might lead to bugs due to the OCR protocol assuming they're pure when they're not

Reply:
I think this logic was maybe fine in OCR2 but in OCR3 seems like a no-no, perhaps we can ask research for help on this.